utils/rowgroup/rowgroup.h

```diff
@@ -1923,11 +1989,13 @@ inline void copyRow(const Row& in, Row* out, uint32_t colCount)
 
   out->setRid(in.getRelRid());
 
-  if (!in.usesStringTable() && !out->usesStringTable())
-  {
-    memcpy(out->getData(), in.getData(), std::min(in.getOffset(colCount), out->getOffset(colCount)));
-    return;
-  }
+  // XXX: this code will copy data incorrectly if sizes of columns differ. Not only for strings, but for eveything.
```

Comment:
I don't mind an additional check but I am strongly against removing memcpy.

Reply:
I suggest to resolve it this way. Move the commented code into a new method Row::copyRowAllColsInline(). What do you think?

Reply:
I'll do that.

But.

Do you really think this can be more than a blip in the program profile?

Reply:
80% percent sure on that.

Reply:
Okay. ;)

Reply:
Okay, I made a copyRowInline with a better memcpy guard check and several lines of comments describing pitfalls to avoid. copyRowInline tries to memcpy and falls back into copyRow when it can't.